src/shogun/distance/Distance.h

```diff
@@ -120,6 +120,29 @@ class CDistance : public CSGObject
 			return distance(idx_a, idx_b);
 		}
 
+		/** NOT IMPLEMENTED
+		 * Precompute squared norms from features of right hand side
+		 * WARNING : Make sure to reset squared norms using reset_squared_norms()
+		 * when features feature matrix are changed.
+		 * Implemented ONLY for Euclidean distance
+		 */
+		virtual void precompute_rhs_squared_norms(){}
+
+		/** NOT IMPLEMENTED
+		 * Precompute squared norms from features of left hand side
+		 * WARNING : Make sure to reset squared norms using reset_squared_norms()
+		 * when features or feature matrix are changed.
+		 * Implemented ONLY for Euclidean distance
+		 */
+		virtual void precompute_lhs_squared_norms(){}
+	
+		/** NOT IMPLEMENTED
+		 * Reset squared norms for features of both sides
+		 * squared norms should be reset whenever features or feature matrix are changed.
+		 * Implemented ONLY for Euclidean distance
+		 */
```

Comment:
Then I can merge it first and this patch here gets smaller. It is already quite large